dashboard/modules/job/job_supervisor.py

```diff
@@ -56,70 +80,614 @@ class JobSupervisor:
     Job supervisor actor should fate share with subprocess it created.
     """
 
+    # Interval of the monitoring loop iterations
+    JOB_MONITOR_LOOP_INTERVAL_S = 1
+    # Interval of logging job status w/o the state changes (allowing
+    # us to track the progress of the monitoring loop)
+    JOB_STATUS_LOG_INTERVAL_S = 600
+    # Timeout to finalize job status after job driver exiting
+    JOB_STATUS_FINALIZATION_TIMEOUT_S = 60
+
+    def __init__(
```

Comment:
Most important changes in JobSupervisor:

 - JS took over whole job management lifecycle from JobManager
 - Actual driver execution have been moved to JobRunner